tests/mfx_dispatch/linux/include/mfx_dispatch_test_hooks.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2017-2018 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef MFX_DISPATCH_HOOKS_H
+#define MFX_DISPATCH_HOOKS_H
+
+#include <mfxvideo.h>
+#include <functional>
+
+struct HookInternalParams
+{
+    mfxVersion emulated_api_version = {{0, 0}};
+};
+
+typedef void* HookHandle;
```

Comment:
Have you thought about returning hook handle or some other pointer/structure which you have created inside the test as mfxSession from MFXInit(Ex)? The idea of the dispatcher is that it should call real API function and pass this handle intact (though it can wrap it in something else inside). I.e. that's something you can check:
MFXInitEx -> you catch dlopen, you catch dlsym and if test really makes init, you catch:
- MFXInitEx and return something in mfxSession
- MFXQueryVersion and can check/work what you have passed as mfxSession
- similar for MFxQueryIMPL

Actually returning different results/statuses from all 3 functions I mentioned above makes a test for MFXInit(Ex).